src/storage-client/src/storage_collections.rs

```diff
@@ -86,38 +86,12 @@ mod metrics;
 pub trait StorageCollections: Debug {
     type Timestamp: TimelyTimestamp;
 
-    /// On boot, reconcile this [StorageCollections] with outside state. We get
-    /// a [StorageTxn] where we can record any durable state that we need.
-    ///
-    /// We get `init_ids`, which tells us about all collections that currently
-    /// exist, so that we can record durable state for those that _we_ don't
-    /// know yet about.
-    ///
-    /// We also get `drop_ids`, which tells us about all collections that we
-    /// might have known about before and have now been dropped.
-    async fn initialize_state(
```

Comment:
Some of these were meant to only be on `StorageCollections` and not actually on `StorageController` anymore. In a lot of cases the storage controller just calls a method on the underlying storage collections but the plan was to move the storage collections out from under the storage controller, and have the controller really be a user of the collections.

An example of that is this method. The storage controller doesn't do anything and just passes through, and actually doesn't need to know about it.